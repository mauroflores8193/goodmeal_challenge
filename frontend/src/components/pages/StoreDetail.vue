<template>
  <div class="flex flex-col h-full text-sm">
    <div class="px-6">
      <Header :title="name" url-back="/"/>
    </div>
    <StoreHeader :icon="icon" :banner="banner"/>
    <div class="px-6 py-2 flex flex-col gap-4">
      <div class="font-medium">Acerca de la tienda</div>
      <div class="flex items-center gap-1">
        <div class="w-6">
          <MarkerIcon class="w-6 h-6"/>
        </div>
        <a href="#" target="_blank" class="text-pink-600 underline leading-6">{{ location }}</a>
      </div>
      <div class="flex items-center gap-1">
        <div class="w-6">
          <ClockIcon class="w-4 h-4 m-auto"/>
        </div>
        <div>Horario de retiro: {{ attentionHours }}</div>
      </div>
      <div class="flex items-center gap-2">
        <div class="font-medium">Calificación</div>
        <StarIcon class="w-4 h-4 text-pink-500"/>
        <div>{{ score }} / 5.0</div>
      </div>
    </div>
    <div class="grid grid-cols-3 gap-x-px gap-y-2 text-sm flex-1 overflow-y-auto px-6">
      <ProductCard v-for="(product, index) in products" :key="index" v-bind="product"/>
    </div>
  </div>
</template>

<script>
import Header from "../molecules/Header.vue";
import StoreHeader from "../organisms/StoreHeader.vue";
import MarkerIcon from '../../assets/images/icons/marker.svg'
import ClockIcon from '../../assets/images/icons/clock.svg'
import StarIcon from '../../assets/images/icons/star.svg'
import ProductCard from "../organisms/ProductCard.vue";

export default {
  name: "StoreDetail",
  components: { ProductCard, StoreHeader, Header, ClockIcon, MarkerIcon, StarIcon },
  data: function () {
    return {
      name: 'GoodMeal Market',
      icon: 'https://upload.wikimedia.org/wikipedia/commons/thumb/a/ab/Logo_TV_2015.png/200px-Logo_TV_2015.png',
      banner: 'https://cdn.pixabay.com/photo/2017/03/22/12/14/easter-nest-2164822_960_720.jpg',
      attentionHours: 'Hoy 09:00 - 21:00 hrs',
      location: 'Av nueva los leones 50, Providencia, Santiago, (Caracol los leones) Zona de delivery Store central',
      latitude: '',
      longitude: '',
      score: 4.4,
      products: [
        { id: 1, name: 'Pack 10 mix almendras pasas y maníi', image: 'https://thumbs.dreamstime.com/z/gourmet-meal-2431449.jpg', price: '4000', discount: 50, realPrice: '8000' },
        { id: 1, name: 'Pack 10 mix almendras pasas y maníi', image: 'http://www.theglutengal.com/wp-content/uploads/2008/09/8-12-08-square-meal.jpg', price: '4000', discount: 50, realPrice: '8000' },
        { id: 1, name: 'Pack 10 mix almendras pasas y maníi', image: 'https://miro.medium.com/max/1400/1*sNhk33H9Te79RJqBuYxBsQ.jpeg', price: '4000', discount: 50, realPrice: '8000' },
        { id: 1, name: 'Pack 10 mix almendras pasas y maníi', image: 'https://thumbs.dreamstime.com/z/plated-pork-main-meal-gourmet-square-white-plate-63563837.jpg', price: '4000', discount: 50, realPrice: '8000' },
        { id: 1, name: 'Pack 10 mix almendras pasas y maníi', image: 'https://www.foodnavigator-usa.com/var/wrbm_gb_food_pharma/storage/images/publications/food-beverage-nutrition/foodnavigator-usa.com/article/2022/01/14/the-meal-plan-from-the-future-squareat-gives-new-meaning-to-square-meals/13156569-1-eng-GB/The-meal-plan-from-the-future-SquarEat-gives-new-meaning-to-square-meals.jpg', price: '', discount: 50 },
        { id: 1, name: 'Pack 10 mix almendras pasas y maníi', image: 'https://media-cldnry.s-nbcnews.com/image/upload/newscms/2021_34/1765653/squareat-inline-02-khu-210825.png', price: '', discount: 50 },
      ]
    }
  }
}
</script>