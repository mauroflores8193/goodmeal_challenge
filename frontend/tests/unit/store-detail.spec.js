import { createLocalVue, mount } from '@vue/test-utils'
import VueRouter from 'vue-router'
import StoreDetail from '@/components/pages/StoreDetail.vue'
import flushPromises from 'flush-promises'
import axios from 'axios'
import store from './../mocks/store.json'

jest.mock('axios')

const localVue = createLocalVue()
localVue.use(VueRouter)
const router = new VueRouter()
test('increments value on click', async () => {
  const mockRoute = {
    params: {
      id: 1
    }
  }
  axios.get.mockResolvedValue({ data: store });
  const tree = mount(StoreDetail, {
    localVue, router, global: {      mocks: { $route: mockRoute }    }, computed: {
      emoticon() {
        return 'ðŸ¤¯'
      },
      bgColor() {
        return 'bg-pink-400'
      }
    }
  })
  await flushPromises();
  expect(tree).toMatchSnapshot();
})
