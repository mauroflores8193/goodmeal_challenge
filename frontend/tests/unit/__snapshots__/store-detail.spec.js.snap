// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`increments value on click 1`] = `
<div class="flex flex-col h-full text-sm">
  <div class="px-6">
    <div class="h-16 flex relative items-center"><a href="#/" aria-current="page" class="absolute left-0 router-link-exact-active router-link-active"></a>
      <h1 class="text-lg font-medium mx-auto">Craig Terry</h1>
    </div>
  </div>
  <div class="relative"><img alt="banner" class="h-18 w-full" src="https://cdn.pixabay.com/photo/2015/10/11/11/20/banner-982162__340.jpg"> <img alt="logo" class="rounded-full absolute -bottom-6 right-8 w-16 h-16 bg-white" src="https://99designs-blog.imgix.net/blog/wp-content/uploads/2018/10/attachment_57810318-e1539290902964.png?auto=format&amp;q=60&amp;fit=max&amp;w=930"> </div>
  <div class="px-6 py-2 flex flex-col gap-4">
    <div class="font-medium">Acerca de la tienda</div>
    <div class="flex items-center gap-1">
      <div class="w-6">
        <!---->
      </div> <a href="https://www.google.com/maps/@-0.0600252813,-20.1813547873,16.5z" target="_blank" class="text-pink-600 underline leading-6">Fuga temporibus dicta temporibus expedita quia.</a>
    </div>
    <div class="flex items-center gap-1">
      <div class="w-6">
        <!---->
      </div>
      <div>Horario de retiro: Hoy 13:34 - 20:34 hrs</div>
    </div>
    <div class="flex items-center gap-1">
      <div class="font-medium">CalificaciÃ³n</div>
      <!---->
      <div>0.7 / 5.0</div>
    </div>
  </div>
  <div class="grid grid-cols-3 gap-x-px gap-y-2 text-sm overflow-y-auto px-6">
    <div class="card p-2 flex flex-col gap-2" stock="44">
      <div class="relative"><img src="https://www.baltimoremagazine.com/wp-content/uploads/2019/05/square-meal-4.jpg" alt="" class="w-full h-20">
        <!---->
      </div>
      <div class="flex gap-1 flex-wrap">
        <div class="price">948.6</div>
        <div class="old-price">9486</div>
      </div>
      <div class="text-white font-medium w-fit px-2 py-1 text-xs rounded-full m-auto bg-pink-400">
        ðŸ¤¯ 10% dcto.
      </div>
      <div class="text-gray-500 text-xs">Filomena Welch I</div>
    </div>
    <div class="card p-2 flex flex-col gap-2" stock="30">
      <div class="relative"><img src="https://img.freepik.com/free-psd/american-food-square-flyer-template_23-2148477325.jpg?w=2000" alt="" class="w-full h-20">
        <!---->
      </div>
      <div class="flex gap-1 flex-wrap">
        <div class="price">446.4</div>
        <div class="old-price">1488</div>
      </div>
      <div class="text-white font-medium w-fit px-2 py-1 text-xs rounded-full m-auto bg-orange-400">
        ðŸ¤© 30% dcto.
      </div>
      <div class="text-gray-500 text-xs">Conor Ryan</div>
    </div>
  </div>
</div>
`;
